src/btreemap/node/v2.rs

```diff
@@ -0,0 +1,305 @@
+
+impl<K: Storable + Ord + Clone> Node<K> {
+    // Loads a V2 node from memory at the given address.
+    pub(super) fn load_v2<M: Memory>(address: Address, page_size: usize, memory: &M) -> Self {
+        // Load the node, including any overflows, into a buffer.
+        let node_buf = read_node(address, page_size, memory);
+
+        // Load the node type.
+        let node_type = match node_buf[NODE_TYPE_OFFSET] {
+            LEAF_NODE_TYPE => NodeType::Leaf,
+            INTERNAL_NODE_TYPE => NodeType::Internal,
+            other => unreachable!("Unknown node type {}", other),
+        };
+
+        // Load the number of entries
+        let num_entries = read_u16_from_slice(&node_buf, NUM_ENTRIES_OFFSET) as usize;
+
+        // Load the entries.
+        let mut keys = Vec::with_capacity(num_entries);
+        let mut encoded_values = Vec::with_capacity(num_entries);
+        let mut offset = ENTRIES_OFFSET;
+        let mut buf = vec![];
+        for _ in 0..num_entries {
+            // Load the key's size.
+            let key_size = read_u32_from_slice(&node_buf, offset) as usize;
+            offset += U32_SIZE;
+
+            // Load the key.
+            buf.resize(key_size, 0);
+            let key = K::from_bytes(Cow::Borrowed(
+                &node_buf[offset.get() as usize..offset.get() as usize + key_size],
+            ));
+            offset += Bytes::from(key_size as u64);
+            keys.push(key);
+        }
+
+        // Load the values
+        for _ in 0..num_entries {
+            // Load the value's size.
+            let value_size = read_u32_from_slice(&node_buf, offset) as usize;
+            offset += U32_SIZE;
+
+            // Load the value.
+            // TODO: Read values lazily.
+            buf.resize(value_size, 0);
+            encoded_values.push(Value::ByVal(
+                node_buf[offset.get() as usize..offset.get() as usize + value_size].to_vec(),
+            ));
+
+            offset += Bytes::from(value_size as u64);
+        }
+
+        // Load children if this is an internal node.
+        let mut children = vec![];
+        if node_type == NodeType::Internal {
+            // The number of children is equal to the number of entries + 1.
+            for _ in 0..num_entries + 1 {
+                let child = address_from_slice(&node_buf, offset);
+                offset += Address::size();
+                children.push(child);
+            }
+
+            assert_eq!(children.len(), keys.len() + 1);
+        }
+
+        let original_overflow_address = address_from_slice(&node_buf, OVERFLOW_ADDRESS_OFFSET);
+
+        Self {
+            address,
+            keys,
+            encoded_values: RefCell::new(encoded_values),
+            children,
+            node_type,
+            version: Version::V2 { page_size },
+            overflow: if original_overflow_address == crate::types::NULL {
+                None
+            } else {
+                Some(original_overflow_address)
```

Comment:
When loading the `Node`, why we do not load the overflow data as well? Is that avoided to decrease refactoring when lazy loading is done or there is another reason?

Reply:
Yes, exactly, we want to do as little work as possible here.